packages/react/src/transactions/adapters/__helpers__/mocks.ts

```diff
@@ -118,3 +124,27 @@ export function mockIMetadataUploader(urlOrError: Url | Error): IMetadataUploade
 
   return uploader;
 }
+
+export function assertUnsignedProtocolCallCorrectness<T extends TransactionRequestModel>(
+  unsignedProtocolCall: UnsignedProtocolCall<T>,
+  broadcastResult: {
+    id: string;
+    typedData: TypedData;
+  },
+) {
+  expect(unsignedProtocolCall.id).toEqual(broadcastResult.id);
+  expect(unsignedProtocolCall.typedData).toEqual(omitTypename(broadcastResult.typedData));
+}
+
+export function assertBroadcastingErrorResultWithRequestFallback(
+  result: Result<unknown, BroadcastingError>,
+  expectedBroadcastingErrorReason: BroadcastingErrorReason,
+  typedData: TypedData,
+) {
+  assertFailure(result);
+  expect(result.error.reason).toEqual(expectedBroadcastingErrorReason);
+  expect(result.error.fallback).toMatchObject({
+    contractAddress: typedData.domain.verifyingContract,
+    encodedData: expect.any(String),
+  });
+}
```

Comment:
I created these helpers to consolidate and simplify test assertions in the coming changes.